frontend/server/src/main/java/org/pytorch/serve/wlm/WorkLoadManager.java

```diff
@@ -150,6 +172,12 @@ public int getNumRunningWorkers(ModelVersionName modelVersionName) {
                 }
                 future.complete(HttpURLConnection.HTTP_OK);
             }
+
+            if (isRestartWorkers) {
+                logger.warn("restarting {} thread(s)", restartNumWorkers);
+                addThreads(threads, model, restartNumWorkers, future);
```

Comment:
Finally after removing all `currentNumWorkers` of workers, add back (i.e., restart) `restartNumWorkers`.  Feel free to have a try with apache benchmark. For example, setting initial `workers` to 1 and scaling up to 5 via `curl -v -X PUT "http://localhost:8081/models/benchmark?min_worker=5"` or setting initial `workers` to 5 and scaling down to 1 `curl -v -X PUT "http://localhost:8081/models/benchmark?min_worker=1"`